src/test/groovy/DataGen.groovy

```diff
@@ -1,15 +1,22 @@
 import com.coder.gateway.models.WorkspaceAgentModel
 import com.coder.gateway.models.WorkspaceAndAgentStatus
 import com.coder.gateway.models.WorkspaceVersionStatus
-import com.coder.gateway.sdk.v2.models.WorkspaceStatus
-import com.coder.gateway.sdk.v2.models.WorkspaceTransition
+import com.coder.gateway.sdk.v2.models.*
 
 class DataGen {
-    static WorkspaceAgentModel workspace(String name, String workspaceName = name) {
+    // Create a random workspace agent model.  If the workspace name is omitted
+    // then return a model without any agent bits, similar to what
+    // toAgentModels() does if the workspace does not specify any agents.
+    // TODO: Maybe better to randomly generate the workspace and then call
+    //       toAgentModels() on it.  Also the way an "agent" model can have no
+    //       agent in it seems weird; can we refactor to remove
+    //       WorkspaceAgentModel and use the original structs from the API?
```

Comment:
👍 to this, I'm assuming we auto-generate the API? I wonder if we could gen the field values from the spec.

Reply:
Oh that would be absolutely wonderful.  The API types appear to be hand-written at the moment.

Reply:
https://github.com/coder/jetbrains-coder/issues/314